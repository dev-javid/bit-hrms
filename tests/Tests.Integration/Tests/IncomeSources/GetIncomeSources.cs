namespace Tests.Integration.Tests.IncomeSources;

public class GetIncomeSources(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
{
    private const string Route = "/api/income-sources?page=1&limit=10";

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public async Task Should_Return_Income_Sources()
    {
        await LoginAsCompanyAdminAsync();
        await FeedDataAsync("Tests/IncomeSources/GetIncomeSources.sql");

        using HttpResponseMessage response = await Client.GetAsync(Route);

        response.StatusCode.Should().Be(HttpStatusCode.OK);
        var result = await response.Content.ReadAsStringAsync();
        Approvals.VerifyJson(result);
    }
}
