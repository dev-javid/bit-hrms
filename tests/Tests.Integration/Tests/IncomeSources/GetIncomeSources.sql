INSERT INTO income_sources (id, company_id, name, description, created_on, created_by, modified_by, modified_on)
VALUES (1,  999, 'Income Source 1', 'Description 1', '2021-01-01', 100, null, null),
       (2,  999, 'Income Source 2', 'Description 2', '2021-01-01', 100, null, null),
       (3,  999, 'Income Source 3', 'Description 3', '2021-01-01', 100, null, null),
       (4,  999, 'Income Source 4', 'Description 4', '2021-01-01', 100, null, null),
       (5,  999, 'Income Source 5', 'Description 5', '2021-01-01', 100, null, null),
       (6,  999, 'Income Source 6', 'Description 6', '2021-01-01', 100, null, null),
       (7,  999, 'Income Source 7', 'Description 7', '2021-01-01', 100, null, null),
       (8,  999, 'Income Source 8', 'Description 8', '2021-01-01', 100, null, null),
       (9,  999, 'Income Source 9', 'Description 9', '2021-01-01', 100, null, null),
       (10, 999,  'Income Source 10', 'Description 10', '2021-01-01', 100, null, null),
       (11, 999,  'Income Source 11', 'Description 11', '2021-01-01', 100, null, null),
       (12, 999,  'Income Source 12', 'Description 12', '2021-01-01', 100, null, null),
       (13, 999,  'Income Source 13', 'Description 13', '2021-01-01', 100, null, null),
       (14, 999,  'Income Source 14', 'Description 14', '2021-01-01', 100, null, null),
       (15, 999,  'Income Source 15', 'Description 15', '2021-01-01', 100, null, null),
       (16, 999,  'Income Source 16', 'Description 16', '2021-01-01', 200, null, null),
       (17, 999,  'Income Source 17', 'Description 17', '2021-01-01', 200, null, null),
       (18, 999,  'Income Source 18', 'Description 18', '2021-01-01', 200, null, null),
       (19, 999,  'Income Source 19', 'Description 19', '2021-01-01', 200, null, null),
       (20, 999,  'Income Source 20', 'Description 20', '2021-01-01', 200, null, null);