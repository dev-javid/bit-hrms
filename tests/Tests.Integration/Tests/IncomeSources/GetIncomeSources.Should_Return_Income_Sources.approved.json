{
    "page": 1,
    "limit": 10,
    "total": 20,
    "items": [
        {
            "id": 20,
            "name": "Income Source 20",
            "description": "Description 20"
        },
        {
            "id": 19,
            "name": "Income Source 19",
            "description": "Description 19"
        },
        {
            "id": 18,
            "name": "Income Source 18",
            "description": "Description 18"
        },
        {
            "id": 17,
            "name": "Income Source 17",
            "description": "Description 17"
        },
        {
            "id": 16,
            "name": "Income Source 16",
            "description": "Description 16"
        },
        {
            "id": 15,
            "name": "Income Source 15",
            "description": "Description 15"
        },
        {
            "id": 14,
            "name": "Income Source 14",
            "description": "Description 14"
        },
        {
            "id": 13,
            "name": "Income Source 13",
            "description": "Description 13"
        },
        {
            "id": 12,
            "name": "Income Source 12",
            "description": "Description 12"
        },
        {
            "id": 11,
            "name": "Income Source 11",
            "description": "Description 11"
        }
    ]
}